pyugrid/ugrid.py

```diff
@@ -770,3 +856,34 @@ def save_as_netcdf(self, filepath):
                 for att_name, att_value in dataset.attributes.items():
                     setattr(data_var, att_name, att_value)
             nclocal.sync()
+
+
+def load_grid(nc, mesh_name=None, load_data=False):
+    """
+    Loads UGrid object from a `netCDF4.Dataset` object or file name/URL,
+    adding the data to the passed-in grid object.
+
+    It will load the mesh specified, or look for the first one it finds if
+    none is specified.
+
+    :param nc: The grid data to be loaded up.
+    :type nc: `netCDF4.Dataset` object or netCDF/URL string
+
+    :param mesh_name=None: name of the mesh to load.
+    :type mesh_name: string
+
+    :param load_data=False: flag to indicate whether you want to load the
+                            associated data or not. The mesh will be loaded
+                            in any case. If False, only the mesh will be
+                            loaded. If True, then all the data associated
+                            with the mesh will be loaded.
+                            This could be huge!
+    :type load_data: boolean
+
+    """
+    if isinstance(nc, Dataset):
```

Comment:
why not:

```
if not isinstance(nc, Dataset):
    nc = Dataset(nc, 'r')
```

or even:

nc = Dataset(nc, 'r') if not isinstance(nc, Dataset)